flink-yarn/src/main/java/org/apache/flink/yarn/YarnApplicationMasterRunner.java

```diff
@@ -326,13 +326,30 @@ protected int runApplicationMaster(Configuration config) {
 
 			// ---- (4) start the actors and components in this order:
 
-			// 1) JobManager & Archive (in non-HA case, the leader service takes this)
-			// 2) Web Monitor (we need its port to register)
+			// 1) Web Monitor (we need its port to register)
+			// 2) JobManager & Archive (in non-HA case, the leader service takes this)
```

Comment:
Why do you change the order of the instantiation? This is not clear to me since you left the original order in the Mesos case.